lib/bolt/util.rb

```diff
@@ -22,6 +22,28 @@ def read_arg_file(file)
         raise Bolt::FileError.new("Error attempting to read #{file}: #{e}", file)
       end
 
+      def read_json_file(path, filename)
```

Comment:
I wonder if we could just alias `read_yaml_hash` and `read_optional_yaml_hash`? Psych syntax errors also seem to be a bit more helpful than JSON's:

```
Psych::SyntaxError ((test.json): did not find expected ',' or '}' while parsing a flow mapping at line 1 column 1)
```

```
JSON::ParserError (784: unexpected token at '{)
```

Reply:
We'd have to refactor those anyway as `.rerun.json` reads and writes an array, not a hash. Those functions are really used for reading configuration files, while the json function is used for internal files (for example we don't read task metadata with it). I don't expect users to even encounter this error. I guess we can load the file with Psych, but there is some level of risk as I'm not entirely sure JSON and YAML are always [guaranteed to be strict supersets](https://yaml.org/spec/1.2/spec.html#id2759572) (for example JSON allows duplicate keys while YAML doesn't), and it generally seems weird to load json with a yaml loader. I don't think it makes sense for the file to be in YAML either, as YAML is really more for data that requires human interaction and I don't expect anyone to ever look at this file.